### ISSUE TITLE
[Feature]: FSDP Actor NCCL timeout is hardcoded to PyTorch default (30 min), not configurable

### ISSUE THREAD

**Issue by BitCalSaul** [RLinf/RLinf#996]
### Feature description

  **Problem**

  The FSDP Actor's default torch.distributed process group is implicitly initialized by   DeviceMesh._get_or_create_default_group(), which calls init_process_group() with no arguments, resulting in   PyTorch's default timeout of 1800 seconds (30 minutes).

  RLinf's MultiChannelProcessGroup has its own configurable timeout (RLINF_TIMEOUT, default 180 min), but   FSDP's internal collective operations (all_gather, reduce, etc.) use the default process group and are not   affected by RLINF_TIMEOUT.

  This causes false timeout kills in two scenarios:
  1. Long training steps (e.g. large gradient accumulation) where a rank's normal computation exceeds 30   minutes between collectives.
  2. Debugging a specific rank (e.g. attaching a debugger to rank 3), which pauses that rank and causes all   other ranks to hit the watchdog timeout.

  WorkNCCL(SeqNum=1878, OpType=_ALLGATHER_BASE, ..., Timeout(ms)=1800000) ran for 1800000 milliseconds before   timing out.

  **Root Cause**

  rlinf/hybrid_engines/fsdp/fsdp_model_manager.py calls create_device_mesh() without first explicitly  initializing torch.distributed. PyTorch's DeviceMesh.__init__ detects this and auto-calls   init_process_group() with the default timeout (1800s), which cannot be overridden externally.

  **Suggested Fix**

  Explicitly call torch.distributed.init_process_group(backend="nccl", timeout=...) before  create_device_mesh() in FSDPModelManager.__init__, with the timeout value configurable via environment   variable or config.

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

**Comment by kakiuwang-ui** [comment]
I looked into this and can confirm it on current `main`. A couple of details that may be useful:

The 30 minutes really does reach every FSDP collective, not just the default group. `DeviceMesh._init_one_process_group` short-circuits when a mesh dimension spans the whole world and returns `_get_default_group()` instead of creating a new group. Since `create_device_mesh` builds `mesh_shape=(world_size,)` with a single `fsdp` dimension, `mesh["fsdp"].get_group() is _get_default_group()` — so the implicit `init_process_group()` inside `init_device_mesh` is what every all-gather and reduce-scatter ends up running on.

There is also no way to work around it from outside the code: `default_pg_timeout` comes from `torch._C._distributed_c10d._DEFAULT_PG_TIMEOUT`, a compiled constant with no environment override.

Reproducible without a GPU, single rank:

```python
os.environ.update(MASTER_ADDR="127.0.0.1", MASTER_PORT="29551",
                  RANK="0", WORLD_SIZE="1", LOCAL_RANK="0")
mesh = init_device_mesh("cpu", mesh_shape=(1,), mesh_dim_names=["fsdp"])
g = mesh["fsdp"].get_group()
print(g is dist.distributed_c10d._get_default_group(),
      g._get_backend(g._device_types[0]).options._timeout)
# True 0:30:00
```

Adding an explicit `init_process_group(timeout=timedelta(minutes=180))` beforehand makes the same probe print `True 3:00:00`.

I have a fix ready and will open a PR shortly. It creates the default group inside `create_device_mesh` using `RLINF_TIMEOUT` rather than adding a new config key, since `ClusterEnvVar.TIMEOUT` is already documented as covering all inter-worker communication. One deviation from the suggested fix: it passes no `backend`, so torch still resolves the per-device backend on its own — hardcoding `"nccl"` would break the Ascend/XPU/MUSA paths that `AcceleratorUtil.get_ccl_backend` supports.

Happy to switch it to a `fsdp_config` key instead if you would rather keep the FSDP timeout separate from the scheduler one.
